presto-docs/src/main/sphinx/sql/select.rst

```diff
@@ -112,6 +122,246 @@ When a ``GROUP BY`` clause is used in a ``SELECT`` statement all output
 expression must be either aggregate functions or columns present in
 the ``GROUP BY`` clause.
 
+**Complex Grouping Operations**
+
+Presto also supports complex aggregations using the ``GROUPING SETS``, ``CUBE``
+and ``ROLLUP`` syntax. This syntax allows users to perform analysis that require
+aggregation on multiple sets of columns in a single query. Complex grouping
+operations do not support grouping on expressions composed of input columns.
+Only column names or ordinals are allowed.
```

Comment:
Oh, my bad. I usually use column ordinals in GROUP BY rather than references so I thought that maybe I can use the output column as a workaround. The output column was actually an expression so I couldn't reference it.